engine/apps/schedules/models/on_call_schedule.py

```diff
@@ -403,6 +402,83 @@ def _merge_events(self, events):
             events = merged
         return events
 
+    def _generate_ical_file_from_shifts(self, qs, extra_shifts=None, allow_empty_users=False):
```

Comment:
`_generate_ical_file_from_shifts` and `preview_shift` methods are moved, unchanged, from Web schedule class to the base class to be reused by Calendar (Terraform) schedules too.